release-assets.yml workflow_dispatch backfill path fails with "resolve-tag-sha.sh: No such file or directory"

> _**Note:** this issue was drafted by Claude (via Claude Code) and posted on my behalf. Verify before triaging. — Ghaith_

## Symptom

Dispatching `Release Assets` manually via `gh workflow run "Release Assets" -f tag=vX.Y.Z` (the path documented in the workflow's input description as "Release tag to backfill") fails immediately:

```
bash: .github/scripts/resolve-tag-sha.sh: No such file or directory
##[error]Process completed with exit code 127.
```

Concrete failure: [run 25434302280](https://github.com/PLC-lang/rusty/actions/runs/25434302280), dispatched against `v0.5.0`. The release was published manually and the assets were uploaded by hand because of this — see notes inline.

## Root cause

The `Resolve source request` step at the top of the job references `bash .github/scripts/resolve-tag-sha.sh ...` to convert the requested tag into a source SHA, but no `actions/checkout` runs before it on the `workflow_dispatch` path. The only checkout in the workflow is `Checkout release commit` further down, which depends on `steps.source.outputs.source_sha` — i.e. the SHA produced by the very step that's failing. Chicken and egg.

The `workflow_run` path doesn't hit this because it gets `head_sha` from the event payload directly:

```yaml
- name: Resolve source request
  run: |
    if [[ "${{ github.event_name }}" == "workflow_run" ]]; then
      ...
      echo "source_sha=${{ github.event.workflow_run.head_sha }}" >> "$GITHUB_OUTPUT"
    else
      TAG="${{ inputs.tag }}"
      SOURCE_SHA=$(bash .github/scripts/resolve-tag-sha.sh ...)   # ← workspace empty
      ...
    fi
```

So **the `workflow_dispatch` backfill path has never worked since it was added** — the `workflow_run` path masks it because it's exercised on every successful master build.

The later script calls (`resolve-tag-sha.sh` at line 140 and the two `find-workflow-run.sh` calls at lines 170-171) all happen *after* `Checkout release commit`, so they're fine.

## Suggested fix

Add an early `actions/checkout@v6` step before `Resolve source request` that pulls just `.github/scripts/` from the default branch, so the resolve script is on disk when needed. The later `Checkout release commit` step then replaces the workspace with the source SHA — that part is unchanged.

```yaml
steps:
  - name: Checkout repository scripts
    uses: actions/checkout@v6
    with:
      sparse-checkout: |
        .github/scripts
      sparse-checkout-cone-mode: false
      fetch-depth: 1

  - name: Resolve source request
    ...
```

`sparse-checkout-cone-mode: false` because the path isn't directory-aligned for cone mode; `fetch-depth: 1` because we don't need any history at this stage.

## Acceptance criteria

- `gh workflow run "Release Assets" -f tag=vX.Y.Z` (manually backfilling assets to an existing release) succeeds and uploads the same set of artifacts as the auto `workflow_run` path would have done.
- The `workflow_run` path (auto-fire after a successful Build Linux/Windows on master) is unchanged.

--- a/.github/workflows/release-assets.yml
+++ b/.github/workflows/release-assets.yml
@@ -22,6 +22,21 @@ jobs:
       actions: read
       contents: write
     steps:
+      # `workflow_dispatch` needs `resolve-tag-sha.sh` on disk before the
+      # `Resolve source request` step runs. The later `Checkout release commit`
+      # step depends on the SHA that step produces, so it cannot do the
+      # checkout itself (chicken-and-egg). Pull just `.github/scripts/` here;
+      # `Checkout release commit` then replaces the workspace with the source
+      # SHA as before.
+      - name: Checkout repository scripts
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v6
+        with:
+          sparse-checkout: |
+            .github/scripts
+          sparse-checkout-cone-mode: false
+          fetch-depth: 1
+
       - name: Resolve source request
         id: source
         env:

